### PR TITLE
Resolve swagger parsing issues

### PR DESCRIPTION
**Pull request by Sheemap**
Resolves https://github.com/odota/core/issues/1703

This doesn't fix the incorrect API definitions, but does fix parsing. So tools can be used to generate clients that are mostly correct.

This PR also adds a new dependency, I'm not sure the guidelines on adding those, but it adds https://www.npmjs.com/package/@apidevtools/swagger-parser in order to parse the spec in a unit test to ensure validity.

### CODE REVIEW

File: routes/spec.js [hunk 1]
```diff
@@ -4481,6 +4495,36 @@ You can find data that can be used to convert hero and ability IDs and other inf
...
+              type: 'array',
+              items: {
+                type: "string",
+              },
+            },
+          },
+        },
+        route: () => '/constants',
+        func: (req, res, cb) => {
+          if (!req.params.resource) {
```
Comment (howardchung) [inline routes/spec.js:4517]: this will never have the resource parameter?

File: package.json [hunk 2]
```diff
@@ -25,6 +25,7 @@
     "url": "http://github.com/odota/core"
   },
   "dependencies": {
+    "@apidevtools/swagger-parser": "^10.0.2",
```
Comment (howardchung) [inline package.json:28]: could be a devdependency?

File: routes/spec.js (hunk 1 above)
Comment (Sheemap) [reply routes/spec.js]: You're right. I cleared out the code that will never be hit

File: package.json (hunk 2 above)
Comment (Sheemap) [reply package.json]: Good point, didn't even think about that. Moved

**Review by howardchung — APPROVED** [review]

**Comment by howardchung** [comment]
unclear why CI is failing--maybe due to the package-lock change?

travis error is:
```
$ sudo bash scripts/build.sh
Error: Cannot perform an interactive login from a non TTY device
The command "sudo bash scripts/build.sh" exited with 1.
```

**Comment by Sheemap** [comment]
I think its actually related to the DOCKER_PASSWORD environment var, its not getting set somehow.

I was able to replicated it locally by setting DOCKER_USERNAME, but leaving DOCKER_PASSWORD unset. With that, the docker login command within the build script fails with that same interactive login error message

Though Im not sure why the master build you recently tried succeeded fine. Is the environment variable dependent on who is running the build? Like I don't have access to the DOCKER_PASSWORD var, so it doesnt set it?

**Comment by Sheemap** [comment]
Yeah, looking more into it, Im almost positive thats the issue.

Your master build output has this near the top

```
Setting environment variables from repository settings
$ export DOCKER_USERNAME=howardc93
$ export DOCKER_PASSWORD=[secure]
$ export STRIPE_SECRET=[secure]
$ export STRIPE_API_PLAN=[secure]
```

While mine only has this
```
Setting environment variables from repository settings
$ export DOCKER_USERNAME=howardc93
```

So for some unknown reason, this build isn't getting the password set.

**Comment by howardchung** [comment]
Oh I believe this is due to security reasons, PRs don't get env vars.

**Comment by Sheemap** [comment]
Ah interesting, I thought package.json was the only file that controlled versions.

I can create a new PR resolving those rules if you'd like. Looks like its a few really simple fixes (and I introduced a few of em : ) . I didn't check to see if linting was a part of the build process

**Pull request merged**